.github/workflows/pr-benchdiff.yml

```diff
@@ -0,0 +1,94 @@
+name: Benchdiff
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  # https://github.community/t/cancelling-rest-of-job-if-condition-is-met/18181
+  trigger:
+    name: Pull request comment trigger
+    outputs:
+      triggered: ${{ steps.output.outputs.triggered }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check comment
+        uses: khan/pull-request-comment-trigger@master
+        id: check
+        with:
+          trigger: '/benchdiff'
+        env:
+          GITHUB_TOKEN: '${{ secrets.GITHUB_TOKEN }}'
+      - name: Set output
+        id: output
+        run: |
+          echo '::set-output name=triggered::${{ steps.check.outputs.triggered }}'
+
+  benchdiff:
+    name: Performance regression check
+    needs: [trigger]
+    if: needs.trigger.outputs.triggered == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Checkout pull request HEAD
+        id: head
+        run: |
+          # Checkout HEAD of pull request and set its sha to step output
+          git fetch origin pull/${{ github.event.issue.number }}/head:PR_HEAD
+          git checkout PR_HEAD
+          echo "::set-output name=head::$(git rev-parse HEAD)"
+
+          # Set modified packages to step output
+          pkgs=$(git diff --name-only ${{ github.sha }} | grep '.go$' | xargs -L1 dirname | sort | uniq)
+          if [ ! -z "${pkgs}" ]; then
+            # awk: Add "./" prefix to let `go test` known they are relative paths
+            # tr: join paths to one line, otherwise benchdiff cannot recognize it
+            pkgs=$(echo "${pkgs}" | awk '{print "./" $0}' | tr '\n' ' ')
+          fi
+          echo "::set-output name=pkgs::${pkgs}"
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - uses: WillAbides/benchdiff-action@v0.3.3
+        id: diff
+        if: steps.head.outputs.pkgs != ''
+        with:
+          benchdiff_version: 0.7.1
+          status_sha: ${{ steps.head.outputs.head }}
+          status_name: Benchdiff result
+          status_on_degraded: neutral
+          benchdiff_args: |
+            --packages="${{ steps.head.outputs.pkgs }}"
+            --count=10
+            --warmup-count=1
+            --benchtime=100000x
+            --benchmem
+            --tolerance=50
+            --base-ref=${{ github.sha }}
+            --debug
```

Comment:
how many seconds this action currently takes?

Reply:
It depends, see argument `--packages`, this action only bench on modifed packages (`git diff <default_branch> <head_of_pr> | grep '.go$' | xargs -L1 dirname | sort | uniq`).